app/src/androidTest/java/com/github/sdpcoachme/database/CachingDatabaseTest.kt

```diff
@@ -96,12 +101,272 @@ class CachingDatabaseTest {
+                    }
+            }.get(5, SECONDS)
+
+        assertTrue(isCorrect)
+    }
+
+    @Test
+    fun sendingMessageForNotCachedChatDoesNotCacheTheChat() {
+        val newMessage = Message(
+            "New Message!",
+            defaultUser.email,
+            LocalDateTime.now().toString(),
+            false
+        )
+
+        val wrappedDatabase = SendMessageDB(defaultChat)
+        val cachingDatabase = CachingDatabase(wrappedDatabase)
+
+        val isCorrect = cachingDatabase.sendMessage(defaultChat.id, newMessage)
+            .thenCompose {
+                cachingDatabase.getChat(defaultChat.id)
+            }.thenApply {
+                assertThat(wrappedDatabase.timesCalled(), `is`(1))
+                assertThat(it, `is`(defaultChat))
+
+                true
+            }.exceptionally {
+                false
+            }.get(5, SECONDS)
+
+        assertTrue(isCorrect)
+    }
+
+    private class MarkMessagesAsReadDB(val defaultChat: Chat): MockDatabase() {
+        var timesCalled = 0
+
+        fun timesCalled() = timesCalled
+        override fun markMessagesAsRead(chatId: String, email: String): CompletableFuture<Void> {
+            return CompletableFuture.completedFuture(null)
+        }
+
+        override fun getChat(chatId: String): CompletableFuture<Chat> {
+            timesCalled++
+            return CompletableFuture.completedFuture(defaultChat)
+        }
+    }
 
+    @Test
+    fun markMessagesAsReadForCachedChatUpdatesCache() {
+        val expectedChat = defaultChat.copy(messages = defaultChat.messages.map {
+            if (it.sender != defaultUser.email)
+                it.copy(readByRecipient = true)
+            else
+                it
+        })
+        val wrappedDatabase = MarkMessagesAsReadDB(defaultChat)
+        val cachingDatabase = CachingDatabase(wrappedDatabase)
+
+        val isCorrect = cachingDatabase.getChat(defaultChat.id) // to place chat into cache
+            .thenCompose {
+                assertThat(wrappedDatabase.timesCalled(), `is`(1))
+
+                cachingDatabase.markMessagesAsRead(defaultChat.id, defaultUser.email)
+                    .thenCompose {
+                        cachingDatabase.getChat(defaultChat.id)
+                            .thenApply {
+                                println("times called ${wrappedDatabase.timesCalled()}")
+                                println("same as expected chat: ${it == expectedChat}")
+
+                                assertThat(wrappedDatabase.timesCalled(), `is`(1))
+                                assertThat(it, `is`(expectedChat))
 
+                                true
+                            }.exceptionally {
+                                false
+                            }
+                    }
+            }.get(5, SECONDS)
 
-    val exampleEmail = "example@email.com"
+        assertTrue(isCorrect)
+    }
+
+    @Test
+    fun sendForNotCachedChatDoesNotCacheTheChat() {
```

Comment:
The name here should rather be "markMessageAsReadForNotCachedChatDoesNotCacheTheChat", no? Otherwise, it is a bit confusing as you already have a test called "sendingMessageForNotCachedChatDoesNotCacheTheChat".

Reply:
Great catch, I will do that!